packages/compiler/src/server/serverlib.ts

```diff
@@ -339,6 +343,113 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function updateImportsOnFileChange(changes: FileEvent[]): Promise<void> {
+    const tspChanges = changes.filter((change) => change.uri.endsWith(".tsp"));
+    const filePathChangedMap = new Map<string, string>();
+
+    if (tspChanges.length === 0 || tspChanges.length % 2 !== 0) {
+      return;
+    }
+
+    // type 1: created, type 3: deleted
```

Comment:
I think there might be a more reliable way to do this the LSP has a way to register this even

<img width="553" alt="image" src="https://github.com/user-attachments/assets/98a2a4c9-b0f9-44c2-9756-1a8575f61ca8" />

`capabilities.workspace.fileOperations.didRename`/``capabilities.workspace.fileOperations.willRename`

Reply:
see svelte implementaiton for exmaple https://github.com/sveltejs/language-tools/blob/58a86babcce68ebb4f5dd51ea03a685c190c5cd3/packages/svelte-vscode/src/extension.ts#L353